http-server/src/server.rs

```diff
@@ -113,26 +112,19 @@ pub struct Server {
 	/// Local address
 	local_addr: Option<SocketAddr>,
 	/// Registered methods.
-	root: RpcModule,
+	methods: Methods,
 	/// Max request body size.
 	max_request_body_size: u32,
 	/// Access control
 	access_control: AccessControl,
 }
 
 impl Server {
-	/// Register a new RPC method, which responds with a given callback.
-	pub fn register_method<F, R>(&mut self, method_name: &'static str, callback: F) -> Result<(), Error>
-	where
-		R: Serialize,
-		F: Fn(RpcParams) -> Result<R, CallError> + Send + Sync + 'static,
-	{
-		self.root.register_method(method_name, callback)
-	}
-
 	/// Register all methods from a module on this server.
-	pub fn register_module(&mut self, module: RpcModule) -> Result<(), Error> {
-		self.root.merge(module)
+	pub fn register_module<Context>(&mut self, module: RpcModule<Context>) -> Result<(), Error> {
+		// TODO: must check for duplicate method names
+		self.methods.extend(module.into_methods());
```

Comment:
I suggest having a separate type `Methods` and have a method `fn merge`

Reply:
`Methods`, being a type alias, can't have `impl`s. :/

Reply:
yeah, I meant `pub struct Methods(FxHashMap<&'static str, Method>)`